connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerTest.java

```diff
@@ -1144,6 +1144,7 @@ public void testAdminConfigsClientOverridesWithAllPolicy() {
         connConfig.put("metadata.max.age.ms", "10000");
 
         Map<String, String> expectedConfigs = new HashMap<>();
+        expectedConfigs.putAll(props);
```

Comment:
The admin configs after overrides will contain both "admin.metadata.max.age.ms", and "metadata.max.age.ms", with different values. I feel like that might be confusing to debug. Could we include just the configs that don't start with "admin." prefix so they would not be overridden?

Reply:
Part of this test is to ensure that overrides happen as expected; leaving out to-be-overridden configs would leave that case uncovered. If anything, it might make sense to expand the tests here to make sure that three distinct sources of configs for the admin client are recognized and handled with the expected precedence. In ascending order, that would be: from the worker config with no prefix, from the worker config with a (stripped) prefix of `admin.`, and from the connector config with a (stripped) prefix of `admin.override.`.

Reply:
Or are you suggesting changes in the actual `Worker` file as opposed to this test?

Reply:
Oh, yeah, the comment was not about the test, just happen to realize this when looking at the testcase. Could we have something like `originalsWithoutPrefix` on `AbstractConfig` and use it to pre-populate `adminProps` with just original configs not containing override ones?

Reply:
Yeah, that should be fine. Probably not worth it to alter the `AbstractConfig` since that's public API but engaging in some effort to make sure that both `admin.foo` and `foo` aren't present in the props given to the admin client seems like a good idea.